src/endpoint/s3/ops/s3_get_bucket_versions.js

```diff
@@ -17,6 +17,11 @@ async function get_bucket_versions(req) {
         throw new S3Error(S3Error.InvalidArgument);
     }
 
+    if (req.query['version-id-marker'] && !req.query['key-marker']) {
+        dbg.warn('A version-id marker cannot be specified without a key marker');
+        throw new S3Error(S3Error.InvalidArgument);
+    }
+
```

Comment:
Maybe worth taking it into a different PR and adding a test for it.

Reply:
sure

Reply:
@v-harihar @liranmauda I don't think it is worth a different PR or test, it's a small change. 

we can have it maybe in another commit in this PR, so we can backport it separately if needed. make sense?

Reply:
sure. Adding the test in another PR in the same commit